terraform/modules/monitoring/grafana.tf

```diff
@@ -0,0 +1,134 @@
+module "grafana" {
+  source                           = "../app"
+  execution_role_arn               = var.execution_role_arn
+  vpc_id                           = var.vpc_id
+  cluster_id                       = aws_ecs_cluster.cluster.id
+  service_name                     = "grafana"
+  subnets                          = var.private_subnets
+  mesh_name                        = var.mesh_name
+  service_discovery_namespace_id   = var.service_discovery_namespace_id
+  service_discovery_namespace_name = var.service_discovery_namespace_name
```

Comment:
One problem with having the module reuse the `service_discovery_namespace_name` created by the `govuk` deployment, is that we cannot destroy and recreate the `govuk` deployment without first destroying the `grafana` deployment.

If the `monitoring/grafana` module has been applied, we'll see the following error when trying `cd govuk-test && terraform destroy`

```
Error: error deleting Service Discovery Private DNS Namespace: ResourceInUse: Namespace has associated services; delete the services before deleting the namespace
```

```
Error: error deleting App Mesh service mesh: ResourceInUseException: Mesh govuk has active VirtualServices, please delete VirtualServices before deleting the Mesh
```

I wonder if we should accept that rebuilding the infrastructure first requires destroying grafana.

Alternatively we could see if we can extricate the grafana deployment completely, or merge it into the govuk deployment. What do you think?

Reply:
Umm.. we wanted to keep it separate from the discussion with Richard. Maybe what we can do it not include an app into the service mesh if the namespace is null. This will solve this issue.

Reply:
🤔 what was the reason for separating it? Perhaps each workspace doesn't need its own grafana? But maybe it does :)

I ask, since it introduces hidden dependencies - e.g. we can't apply the main branch now (https://cd.gds-reliability.engineering/teams/govuk-tools/pipelines/deploy-apps-test/jobs/run-terraform/builds/71). If we put the monitoring stuff back into the govuk deployment it'll mean extra grafanas but I think that's OK - we shouldn't be designing the terraform for production around a tool for the test environment.

Reply:
I chatted with Richard about the motivation for splitting them up. The main reason is that we can apply grafana to production without having to bring up a whole cluster, and get immediate benefit from the new version of grafana (early gains from replatforming 😄).

We could create a `monitoring` ECS Cluster, and not use App Mesh for this app, since it just needs a public ALB. That would remove some dependencies on the govuk module. What do you think?

Reply:
Now the monitoring cluster is independent of govuk one.